### DISCUSSION TITLE
Change position and bends on link vertices are per user intention in a directed layout

### DISCUSSION THREAD

**Discussion by jibinjoy8897** [clientIO/joint#2536] [Q&A]
### Introduction

I would like to be able to move the link line in all directions
Currently, the link lines are automatically drawn when we move the nodes.
All drawing must be done according to human intention (should be able to adjust the link shape, bends etc...)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Could you elaborate please? Perhaps providing images help us understand.

**Reply by jibinjoy8897** [reply]
![LinkAdjust](https://github.com/clientIO/joint/assets/56633789/61059aa9-5c14-40b8-8765-9368f6274a3f)

I would like to be able to drag the link line in the direction of the red line in the figure.Currently only link length changes when nodes are moved

**Reply by kumilingus** [reply]
If you want the link to stay orthogonal, you need to use an orthogonal router.

In case of `DirectedGraph` you can use the `rightAngle` router.
```js
// If `rankDir` is `"TB"`
// if `useVertices` is `true`
graph.getLinks().forEach(link => {
    link.router('rightAngle', { useVertices: true });
    link.prop('source/anchor', { name: 'bottom' }, { rewrite: true });
    link.prop('target/anchor', { name: 'top' }, { rewrite: true });
});
```

In general, however, this is a complex issue and the result generated by our router may differ from what you would expect from `DirectedGraph`.

You would like to apply the same logic as used in `DirectedGraph` but with custom nodes positions. `DirectedGraph` (`Dagre`) does not support it (neither we).
